src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Requests/DataSetGetQueryRequest.cs

```diff
@@ -0,0 +1,102 @@
+using System.ComponentModel;
+using FluentValidation;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Validators;
+
+namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Requests;
+
+public record DataSetGetQueryRequest
+{
+    /// <summary>
+    /// Query criteria relating to filter options.
+    /// </summary>
+    public DataSetGetQueryFilters? Filters { get; init; }
+
+    /// <summary>
+    /// Query criteria relating to geographic levels.
+    /// </summary>
+    public DataSetGetQueryGeographicLevels? GeographicLevels { get; init; }
+
+    /// <summary>
+    /// Query criteria relating to location options.
+    /// </summary>
+    public DataSetGetQueryLocations? Locations { get; init; }
+
+    /// <summary>
+    /// Query criteria relating to time periods.
+    /// </summary>
+    public DataSetGetQueryTimePeriods? TimePeriods { get; init; }
+
+    /// <summary>
+    /// The IDs of indicators to return values for.
+    /// </summary>
+    public required IReadOnlyList<string> Indicators { get; init; }
```

Comment:
Maybe a `HashSet` might be better?

Reply:
Unfortunately this would lead to loss of ordering information, which is needed for the validator to provide the correct path to the item that failed validation.

We do any deduplication down in the service layer before any queries actually get executed anyway.

Reply:
Ahhhh, you're completely right.